test/common.js

```diff
@@ -204,12 +204,48 @@ Session.create = function create(scenario) {
   return new Session({ scenario: scenario });
 };
 
-Session.loadCore = function loadCore(executable, core, ranges) {
-  return new Session({
+exports.saveCore = function saveCore(options, cb) {
+  const scenario = options.scenario;
+  const core = options.core || exports.core;
+
+  // Create a core and test
+  const sess = Session.create(scenario);
+  sess.timeoutAfter(exports.saveCoreTimeout);
+
+  sess.waitBreak(() => {
+    sess.send(`process save-core ${core}`);
+    // Just a separator
+    sess.send('version');
+  });
+
+  sess.wait(/lldb-/, (err) => {
+    if (err) {
+      return cb(err);
+    }
+    sess.quit();
+
+    if (!process.env.LLNODE_NO_RANGES) {
+      const ranges = core + '.ranges';
+      exports.generateRanges(core, ranges, cb);
+    } else {
+      cb(null);
+    }
```

Comment:
Can you swap the branches?  No negation is never less easy to parse for humans, if you catch my drift.